storage/innobase/include/srv0srv.h

```diff
@@ -129,6 +129,33 @@ struct srv_stats_t {
 	ulint_ctr_1_t		lock_deadlock_count;
 
 	ulint_ctr_1_t		n_lock_max_wait_time;
+
+	/** number of contiguous page reads */
+	ulint_ctr_64_t		scan_pages_contiguous;
+	/** number of jumpy page reads */
+	ulint_ctr_64_t		scan_pages_jumpy;
+	/** total jump distance performed by the latest query */
+	ulint_ctr_64_t		scan_pages_total_jump_distance;
+	/** amount of data in all pages read by the latest query */
+	ulint_ctr_64_t		scan_data_in_pages;
+	/** amount of garbage in all pages read by the latest query */
+	ulint_ctr_64_t		scan_garbages_in_pages;
+
+	/** resets scan_pages_contiguous, scan_pages_jumpy, scan_data_in_pages
+	and scan_garbages_in_pages */
+	void reset_scan_stats()
+	{
+		scan_pages_contiguous.sub(
```

Comment:
Resetting to zero instead of substracting would be more natural and more efficient (and less racy in some cases, but that's dependent on the general concurrency design for these counters)